### PR TITLE
Augment Tags to EMR

### PR DESCRIPTION
**Pull request by salwandhruvdev**
@kapilt This adds tags to EMR resource. Awaiting your suggestions. 
Thanks.

### CODE REVIEW

File: c7n/resources/emr.py [hunk 1]
```diff
@@ -84,9 +84,13 @@ def consolidate_query_filter(self):
         return result
 
     def augment(self, resources):
+        client = local_session(self.get_resource_manager('emr').session_factory).client('emr')
         # remap for cwmetrics
         for r in resources:
             r['ClusterId'] = r['Id']
+            cluster = client.describe_cluster(ClusterId=r['ClusterId'])
+            resource_tags = cluster['Cluster']['Tags']
```
Comment (kapilt) [inline c7n/resources/emr.py]: describe emr actually has quite a bit of other useful information in it thats not in the listing, i suggest we discard the list result and replace the resource with the describe output. i need this as well for some of the emr luks checks which need securityconfiguration which isn't in the list output.

**Comment by kapilt** [comment]
also needs to update permissions in use on the resource manager for the describe call

**Comment by salwandhruvdev** [comment]
@kapilt Added mark-for-op, remove-tag and marked-for-op to EMR. Awaiting your review. Thanks for your help.

**Comment by salwandhruvdev** [comment]
Thanks for all the suggestions @kapilt . I have integrated flake8 with IDE and made the modifications. Ready to merge.

**Pull request merged**